expression/util.go

```diff
@@ -526,3 +527,57 @@ func GetParamExpression(ctx sessionctx.Context, v *driver.ParamMarkerExpr, useCa
 	}
 	return value, nil
 }
+
+// ParamToByItemNode generate ByItem node from ParamMarkerExpr.
+func ParamToByItemNode(ctx sessionctx.Context, v *driver.ParamMarkerExpr) (*ast.ByItem, bool, error) {
+	value, err := GetParamExpression(ctx, v)
+	if err != nil {
+		return nil, true, errors.Trace(err)
+	}
+	str, isNull, err := GetStringFromConstant(ctx, value)
+	if err != nil {
+		return nil, true, errors.Trace(err)
+	}
+	if isNull {
+		return nil, true, nil
+	}
+	pos, err := strconv.Atoi(str)
+	if err == nil {
+		byItem := &ast.ByItem{Expr: &ast.PositionExpr{N: pos, P: v}}
+		return byItem, false, nil
+	}
+	return nil, true, nil
+}
+
+// GetStringFromConstant gets a string value from the Constant expression.
+func GetStringFromConstant(ctx sessionctx.Context, value Expression) (string, bool, error) {
+	con, ok := value.(*Constant)
+	if !ok {
+		err := errors.Errorf("Not a Constant expression %+v", value)
+		return "", true, errors.Trace(err)
+	}
+	str, isNull, err := con.EvalString(ctx, chunk.Row{})
+	if err != nil {
```

Comment:
ditto

Reply:
The updated PR should tiny up the codes.